src/context.rs

```diff
@@ -67,18 +68,26 @@ impl Context {
         let inner_ctx =
             NonNull::new(unsafe { ibv_open_device(*dev) }).ok_or_else(io::Error::last_os_error)?;
         unsafe { ibv_free_device_list(dev_list_ptr) };
-        let mut gid = Gid::default();
-        let mut errno =
-            unsafe { ibv_query_gid(inner_ctx.as_ptr(), port_num, gid_index.cast(), gid.as_mut()) };
-        if errno != 0_i32 {
-            error!(
-                "open_device, err info : {:?}",
-                io::Error::from_raw_os_error(0_i32.sub(errno))
-            );
-            return Err(io::Error::from_raw_os_error(0_i32.sub(errno)));
-        }
+
+        let gid = {
+            let mut gid = MaybeUninit::<ibv_gid>::uninit();
+            let gid_index = gid_index.cast();
+            // SAFETY: ffi
+            let errno =
+                unsafe { ibv_query_gid(inner_ctx.as_ptr(), port_num, gid_index, gid.as_mut_ptr()) };
+            if errno != 0_i32 {
+                assert_ne!(errno, i32::MIN);
```

Comment:
What's purpose of this assertion?

Reply:
<https://doc.rust-lang.org/stable/std/primitive.i32.html#method.wrapping_neg>
It seems unnecessary. I will remove it.

Reply:
The possible return value is 0 or -1, we won't i32::MIN here. Ref: https://www.rdmamojo.com/2012/08/02/ibv_query_gid/